modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/AnnotationsBasedDescriber.java

```diff
@@ -244,9 +245,9 @@ private void calculateExtendedTypes(Class<?> actingClass, Method method, Operati
             extensionOf = actingClass.getAnnotation(ExtensionOf.class);
         }
 
-        if (extensionOf != null)
+        if (extensionOf != null || extensionType.getAnnotation(Extensible.class) != null)
         {
-            operation.withCapability(new ExtendingOperationCapability(extensionOf.value()));
+            operation.withCapability(new ExtendingOperationCapability(extensionOf != null ? extensionOf.value() : extensionType));
```

Comment:
this if block is kinda hard to follow, not to mention that per each operation you'll be evaluation extensionOf != null twice (once in the if and again in the ternary).

Do you think it'd be more clear if:

1) Check for extensionType to be extensible in the describer's constructor and store it on a private final boolean field
2) break this down in if (extensionOf != null ) {..} else if(this.isExtensible) {... }

?